docs/config/server-options.md

```diff
@@ -89,9 +89,9 @@ In some cases, you might also want to configure the underlying dev server (e.g.
 export default defineConfig({
   server: {
     proxy: {
-      // string shorthand
+      // string shorthand, matches calls to '/foo/bar' by default
       '/foo': 'http://localhost:4567',
-      // with options
+      // with options, returns http://jsonplaceholder.typicode.com/bar
```

Comment:
> the matching subsequent folders by default is what was confusing me. I thought I needed a rewrite to achieve this.

If this is the reason, I think it's better to update the description (line 80) because these examples are a bit ambiguous.